library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -661,6 +672,7 @@ public void emitTuples()
           if (line == null) {
             LOG.info("done reading file ({} entries).", offset);
             closeFile(inputStream);
+            fileClosed = true;
```

Comment:
Thank you for pointing it out. It does help me identify a bug when the current window finishes reading all the data, and closeFile() is called in next window by itself only. I've updated the fix, and it also covers the case for empty file. I've added both unit test cases.